app/Models/Dataset.php

```diff
@@ -478,9 +478,37 @@ public function json()
             'features' => $features,
             'display' => $featureCollectionConfig->toArray(),
         );
+      
+        if( count($features) == 0){
+            $dataset->description = null; 
+            $allfeatures['metadata']['warning'] = "0 results found";

```

Comment:
This should be written as appending instead of replacing as the layer might have its own warning message.